src/vendor/wordpress/styles/_core-blocks.scss

```
@@ -138,32 +124,53 @@ $wp-button-gap: size.$spacing-gap-button-group-default;
   }
 }
 
-/**
- * Gutenberg block: Image
- * Styles for images inserted via Gutenberg blocks.
- *
- * 1. Since we are applying `margin-top` to CMS content via our o-rhythm
- *    pattern, the default `margin-bottom` value on Gutenberg image blocks is
- *    unnecessary, so we remove it here.
- */
+/// Gutenberg block: Image
+/// Styles for images inserted via Gutenberg blocks.
+
 .wp-block-image {
-  margin-block-end: 0; /* 1 */
   position: relative;
 
-  /**
-   * Setting a width of `100%` prevents resized and centered images from
-   * breaking layout when placed after a right-aligned image in a post.
-   */
-  &.aligncenter {
-    inline-size: 100%;
+  /// Since we are applying `margin-top` to CMS content via our o-rhythm
+  /// pattern, the default `margin-bottom` value on Gutenberg figcaption
+  /// elements is unnecessary, so we remove it here.
+  &,
+  figcaption {
+    margin-block-end: 0;
   }
 
-  /* Since we are applying `margin-top` to CMS content via our o-rhythm
-   * pattern, the default `margin-bottom` value on Gutenberg figcaption
-   * elements is unnecessary, so we remove it here.
-   */
+  /// When some alignment options are selected, WordPress nests the figure with
+  /// the alignment class within the `wp-block-image` element.
+  ///
+  /// 1. The default top margin makes these alignment options appear misaligned
+  ///    unless they are floating midway through text, which is extremely
+  ///    uncommon.
+  /// 2. We constrain the width to 50% minus half of WordPress's default gutter
+  ///    size. This makes the adjacent content feel visually balanced.
+  .alignleft,
+  .alignright {
+    margin-block-start: 0; // 1
+    max-inline-size: calc(50% - 0.5em); // 2
+  }
+}
+
+/// Because WordPress changes the structure of the Image block markup depending
+/// on the alignment option but does not append the alignment class to the
+/// parent element in all cases, this was the only solution I could find for
+/// targeting its _default_ state. We may be able to simplify once the `:has`
+/// selector is better supported!
+figure.wp-block-image {
+  /// When this element has no alignment option set, attempt to fill the
+  /// container padding.
+  &:not(.alignwide):not(.alignfull) {
```

```suggestion
  &:not(.alignwide, .alignfull) {
```

You can simplify this a bit, since `:not` accepts a comma-separated list of selectors